cli/internal/scope/scope.go

```diff
@@ -169,8 +171,21 @@ func (o *Opts) getPackageChangeFunc(scm scm.SCM, cwd string, packageInfos map[in
 	}
 }
 
-func repoGlobalFileHasChanged(opts *Opts, changedFiles []string) (bool, error) {
-	globalDepsGlob, err := filter.Compile(opts.GlobalDepPatterns)
+func getDefaultGlobalDeps(packageManager *packagemanager.PackageManager) []string {
+	// include turbo.json, root package.json, and root lockfile as implicit global dependencies
+	defaultGlobalDeps := []string{
+		"turbo.json",
+		"package.json",
+	}
+	if packageManager != nil {
+		defaultGlobalDeps = append(defaultGlobalDeps, packageManager.Lockfile)
```

Comment:
I think we eventually want more sophisticated handling on the lockfile. See #2162

Reply:
Good call, I'll add a todo here